testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/page/ApplicationsPage.java

```diff
@@ -17,14 +17,46 @@
 
 package org.keycloak.testsuite.ui.account2.page;
 
+import org.keycloak.representations.account.ClientRepresentation;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * @author Vaclav Muzikar <vmuzikar@redhat.com>
  */
 public class ApplicationsPage extends AbstractLoggedInPage {
+    @FindBy(xpath = "//ul[@id='applications-list']/li/div/div[@class='pf-c-data-list__item-content']")
+    private List<WebElement> applications;
+
     @Override
     public String getPageId() {
         return "applications";
     }
 
+    public List<ClientRepresentation> getApplications() {
+        ArrayList<ClientRepresentation> apps = new ArrayList<>();
+        for(WebElement app : applications) {
+            apps.add(toRepresentation(app));
+        }
+        return apps;
+    }
+
+    private ClientRepresentation toRepresentation(WebElement app) {
+        ClientRepresentation client = new ClientRepresentation();
+        String clientName = app.findElement(By.xpath("(div)[1]")).getText();
+        client.setClientName(clientName);
+        boolean userConsentRequired = !app.findElement(By.xpath("(div)[2]")).getText().equals("Internal");
```

Comment:
Please use [`getTextFromElement`](https://github.com/keycloak/keycloak/blob/f426643225337f17f640d43e0f5b236f09d70e9d/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/util/UIUtils.java#L159) to avoid any browser-related problems in the future.